includes/wc-term-functions.php

```diff
@@ -395,11 +395,21 @@ function wc_reorder_terms( $the_term, $next_id, $taxonomy, $index = 0, $terms =
 		// set order
 		$index++;
 		$index = wc_set_term_order( $term->term_id, $index, $taxonomy );
+		
+		/**
+		 * After a term has had it's order set.
+		*/
+		do_action( 'woocommerce_set_term_order', $term, $index, $taxonomy );
 
 		// if that term has children we walk through them
 		$children = get_terms( $taxonomy, "parent={$term->term_id}&menu_order=ASC&hide_empty=0" );
 		if ( ! empty( $children ) ) {
 			$index = wc_reorder_terms( $the_term, $next_id, $taxonomy, $index, $children );
+			
+			/**
+			 * After a term has had its children reordered.
+			*/
+			do_action( 'woocommerce_reorder_terms', $the_term, $next_id, $taxonomy, $index, $children );
```

Comment:
So maybe it should... The function is ran on all children, so it then runs in all cases. Like I mentioned on one of the others, actions should be general purpose. Not tailored to one user's edge case.